### PR TITLE
Adapt to C++ changes in libecl

### PR DESCRIPTION
**Pull request by joakim-hove**
Adapt to C++ changes in libecl summary nodes: https://github.com/Statoil/libecl/pull/538

I intend to merge this - along with the libecl PR on friday 30.th of november. The two PR must be merged together.

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this libecl=538 please

**Comment by joakim-hove** [comment]
jenkins build this libecl=538 please

**Comment by joakim-hove** [comment]
jenkins build this libecl=538 please

File: src/opm/output/eclipse/Summary.cpp [hunk 1]
```diff
@@ -1180,68 +1163,45 @@ Summary::Summary( const EclipseState& st,
...
+        std::string keyword = node.keyword();
 
-        const auto single_value_pair = single_values_units.find( keyword );
-        const auto funs_pair = funs.find( keyword );
-        const auto region_pair = region_units.find( keyword );
-        const auto block_pair = block_units.find( keyword );
+        const auto single_value_pair = single_values_units.find( keyword.c_str() );
+        const auto funs_pair = funs.find( keyword.c_str() );
+        const auto region_pair = region_units.find( keyword.c_str() );
+        const auto block_pair = block_units.find( keyword.c_str() );
```
Comment (bska) [inline src/opm/output/eclipse/Summary.cpp]: Why do you use `.c_str()` here?  These maps all use `std::string` as their key types.

**Comment by joakim-hove** [comment]
jenkins build this libecl=538 please

File: src/opm/output/eclipse/Summary.cpp (hunk 1 above)
Comment (joakim-hove) [reply src/opm/output/eclipse/Summary.cpp]: > Why do you use .c_str() here? These maps all use std::string as their key types.

I will update - I just tried to limit the amount of changes

**Comment by joakim-hove** [comment]
jenkins build this libecl=538 please

**Comment by joakim-hove** [comment]
jenkins build this libecl=538 please

**Comment by joakim-hove** [comment]
jenkins build this libecl=538 please

**Comment by joakim-hove** [comment]
jenkins build this libecl=538 please

**Comment by joakim-hove** [comment]
jenkins build this libecl=538 please

**Comment by joakim-hove** [comment]
jenkins build this libecl=538 please

**Comment by joakim-hove** [comment]
jenkins build this libecl=538 please

**Pull request merged**